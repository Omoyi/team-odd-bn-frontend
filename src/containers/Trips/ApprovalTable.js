import React, { Component } from 'react';
import { connect } from 'react-redux';
import '../../assets/css/table.scss';
import { Redirect } from 'react-router-dom';
import AvailRequestsView from '../../views/trips/ManagerView';
import Dashboard from '../../views/Dashboard/sidebar';
import availRequests, { updateSpinnerStatus } from '../../redux/actions/tripsActions/availRequestsActions';
import verifyToken from '../../helpers/verifyToken';

class ApprovalRequests extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  async componentDidMount() {
    const { availRequests, updateSpinnerStatus } = this.props;
    await updateSpinnerStatus();
    await availRequests();
  }

  render() {
    const { availRequests } = this.props.stateObject;
    const data = availRequests && availRequests.allRequests && availRequests.allRequests.data;
    return (
      <>
        <Dashboard>
          { verifyToken(localStorage.getItem('token')).roleId === 6 ? <AvailRequestsView data={data} /> : <Redirect to="/requests" /> }
          <AvailRequestsView data={data} />
        </Dashboard>
      </>
    );
  }
}

export const mapStateToProps = (state) => ({
  stateObject: state.trips,
});

export default connect(mapStateToProps, { availRequests, updateSpinnerStatus })(ApprovalRequests);
