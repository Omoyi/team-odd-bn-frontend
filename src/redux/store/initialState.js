
const state = {
  search: {
    key: '',
    payload: '',
  },
  popUpsDisplay: {
    deleteComment: 'none',
    editTrip: 'none',
    currentPopUp: 'editTrip',
  },
  pagination: {
    pageNo: 0,
    itemsPerPage: 5,
  },
  accommodation: {
    createAccommodation: {
      rooms: [],
      photos: [],
    },
  },
  auth: {
    signup: {
      spinnerStatus: false,
      signupInputData: {},
      signupData: {},
      signupError: {},
    },
    signin: {
      spinnerStatus: false,
      signinInputData: {},
      signinData: {},
      signinError: {},
    },
  },
  allUsers: {
    paginatedRequest: [],
    paginationStart: 0,
    paginationEnd: 5,
  },
  trips: {
    tripRequests: {
      getCity: [],
      onewayData: {},
      onewayError: {},
      onewayInput: {},
    },
    availRequests: {
      approveRequestMessage: {},
    },
    requests: {
      requestsData: [],
      singleRequestData: {},
      requestCommentsData: {},
      requestsError: {},
      paginatedRequest: [],
      paginationStart: 0,
      paginationEnd: 5,
    },
    getStats: {
      getStatsInput: {
        tripType: 1,
      },
      getStatsData: {},
      getStatsError: {},
      tripStatsCounter: {},
    },
  },
  viewProfile: {
    profile: {},
  },
  requestsData: {
    data: '',
  },
};
export default state;
