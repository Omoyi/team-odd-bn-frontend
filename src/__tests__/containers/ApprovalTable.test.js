import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow, mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import ApprovalTable from '../../containers/Trips/ApprovalTable';
import initialState from '../../redux/store/initialState';


const mockStore = configureStore([]);
let wrapper;
let store;
describe('Approval Table Test Suite ', () => {
  it('Test snapshot using the initial state', () => {
    store = mockStore(initialState);
    wrapper = mount(
      <Provider store={store}>
        <Router>
          <ApprovalTable availRequests={jest.fn()} updateSpinnerStatus={jest.fn()} />
        </Router>
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
