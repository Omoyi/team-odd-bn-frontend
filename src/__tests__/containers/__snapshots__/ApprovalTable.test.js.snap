// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Approval Table Test Suite  Test snapshot using the initial state 1`] = `ReactWrapper {}`;
